src/key.rs

```diff
@@ -0,0 +1,137 @@
+use crossterm::event::{KeyCode, KeyModifiers};
+use serde::{de, Deserialize};
+
+#[derive(Debug, Deserialize, Default)]
+#[serde(default)]
+pub struct KeyMap {
+    #[serde(deserialize_with = "deseralize_key")]
+    pub remove_previous_word: Option<Key>,
```

Comment:
I think these should be named `delete_word` and `delete_char`. Ctrl-Backspace usually removes the current word.

Reply:
usually Ctrl+Backspace, removes the word (part of the word if the cursor is between the word) before the cursor, that's why i chose it to name it remove_previous_word. In my opinion `delete_word` would somewhat be misleading, because `delete_word` indicates the its deleting the whole word, but instead it removes the word before the cursor.